src/registry/domain/options-sanitiser.js

```diff
@@ -53,11 +53,38 @@ module.exports = function(input) {
   }
 
   options.customHeadersToSkipOnWeakVersion = (options.customHeadersToSkipOnWeakVersion ||
-    [])
-    .map(s => s.toLowerCase());
+    []
+  ).map(s => s.toLowerCase());
 
   options.port = process.env.PORT || options.port;
   options.timeout = options.timeout || 1000 * 60 * 2;
 
+  if (options.s3) {
+    options.storage = {};
+    options.storage.adapterType = 's3';
+    options.storage.options = options.s3;
+  }
+
+  if (options.refreshInterval && options.storage) {
+    options.storage.options.refreshInterval = options.refreshInterval;
+  }
+
+  if (options.verbosity && options.storage) {
+    options.storage.options.verbosity = options.verbosity;
+  }
+  if (
+    options.storage &&
+    options.storage.adapterType &&
+    !options.storage.adapter
+  ) {
+    switch (options.storage.adapterType) {
```

Comment:
I think we could simply check for `options.s3` here without the need for an adapterType. 
In the case that `s3` is not avail, you will need to declaratively pass an adapter and options and rely on the new API. What you think?

Reply:
My main reason for using adapterType was the the user would have to specify the following:

adapter: require('oc/src/domain/registry/s3')

It just didn't seem elegant to have people reach down into the source to pull back config info.  With that I thought about removing the adapter from the config, but I also liked how I could add my own if I had a proprietary internal datastore that I didn't want to have in open source land.

Reply:
Ok, I see what you trying to do. 
So, if we are using `adapterType` mainly for the default adapter (`s3`) to avoid people to reach down to get the default adapter we could just make it optional? if you don't specify an adapter it will use the s3 one that come with oc, otherwise you need to pass your own? This way we don't add an option that is actually only been used for s3.